packages/main/src/plugin/kubernetes-context-state.ts

```diff
@@ -400,14 +427,44 @@ export class ContextsManager {
           console.error(`informer connect error on path ${path} for context ${context.name}: `, String(err));
           options.onConnectionError?.(String(err));
         }
-        options.onReachable?.(err === undefined);
+        if (err) {
+          this.setReachableNow(options, false);
+        } else {
+          this.setReachableDelay(options, true);
+        }
       });
     }
     this.restartInformer<T>(informer, context, options);
     return informer;
   }
 
-  private restartInformer<T>(
+  private setReachableDelay<T extends KubernetesObject>(options: CreateInformerOptions<T>, reachable: boolean): void {
+    clearTimeout(options.connectionDelay);
+    options.connectionDelay = setTimeout(() => {
+      this.doSetReachable(options, reachable);
+    }, connectTimeout);
+  }
+
+  private setReachableNow<T extends KubernetesObject>(options: CreateInformerOptions<T>, reachable: boolean): void {
+    clearTimeout(options.connectionDelay);
+    options.connectionDelay = undefined;
+    this.doSetReachable(options, reachable);
+  }
+
+  private doSetReachable<T extends KubernetesObject>(options: CreateInformerOptions<T>, reachable: boolean): void {
+    options.onReachable?.(reachable);
+    if (reachable) {
+      this.resetConnectionAttempts(options);
+    }
+  }
+
+  private resetConnectionAttempts<T extends KubernetesObject>(options: CreateInformerOptions<T>): void {
+    options.backoff.reset();
+    clearTimeout(options.timer);
+    options.timer = undefined;
```

Comment:
would it be better using disposable ?

```
options.dispose();
```

Reply:
As there are several timers in options, it would not work here. But I note that these timers need to be disposed when podman desktop exits, I created an issue to track this (#6198)